form/src/main/java/org/dhis2/form/data/DataEntryBaseRepository.kt

```diff
@@ -40,8 +41,20 @@ abstract class DataEntryBaseRepository(
         val item = when {
             fieldUiModel.optionSet != null -> {
                 fieldUiModel.apply {
-                    this.optionsToHide = listOf(optionsToHide, optionsInGroupsToHide).flatten()
-                    this.optionsToShow = optionsInGroupsToShow
+                    val optionsetConfig = optionSetConfiguration
+                    this.optionSetConfiguration = when (optionsetConfig) {
+                        is OptionSetConfiguration.BigOptionSet -> optionsetConfig.copy(
+                            optionsToHide = listOf(optionsToHide, optionsInGroupsToHide).flatten(),
+                            optionsToShow = optionsInGroupsToShow
+                        )
+                        is OptionSetConfiguration.DefaultOptionSet -> optionsetConfig.copy(
+                            optionsToHide = listOf(optionsToHide, optionsInGroupsToHide).flatten(),
+                            optionsToShow = optionsInGroupsToShow
+                        )
```

Comment:
haven't these two branches the same return object?. It looks like you differentiate between optionsetConfig != null or not. Maybe you could do something like:
`this.optionSetConfiguration = optionsetConfig?.let {
optionsetConfig.copy(
                            optionsToHide = listOf(optionsToHide, optionsInGroupsToHide).flatten(),
                            optionsToShow = optionsInGroupsToShow
                        )
} ?: optionSetConfiguration`